### DISCUSSION TITLE
Blue verified checkmark at Flathub

### DISCUSSION THREAD

**Discussion by alexisguerreroabg** [helge17/tuxguitar#563] [General]
Hello, thank you so much for keeping this project.
I use Flathub's version, and I once consulted in your forum on how the apps are listed as "verified," to which I was told the following "They will need to verify them on their own, we don’t handle that for each app. There is a process for authors to enable that."

You may be able to get it, I know it's something of little relevance, but before new users you can give them more confidence to install the app.

Greetings from Chile.

**Comment by guiv42** [comment]
Hi. Sorry, but I don't really understand: I don't know what a "verified" app is, and I don't understand what you refer to with "your forum"?
Possibly this repo here may not be the right one for such a request. Here, we try to maintain the source code of the app, and we deliver installable builds (for Linux: .deb .rpm and .tgz). Formally there is no link with the flatpak version, which is built independently in [another github repository](https://github.com/flathub/ar.com.tuxguitar.TuxGuitar), not under our control.

**Comment by alexisguerreroabg** [comment]
Sorry for my bad English. I am not really fluent in the language.
I was referring to the Flathub forum.
Regarding verified applications, this site explains it better than me. 
https://docs.flathub.org/docs/for-users/verification/
I thought the flatpak version was under your supervision, sorry.
Regards.

**Reply by guiv42** [reply]
No need to apologize, there's no problem.
Thanks for the link, I'll have a look. Flathub is something completely independent, but eventually we could try to contribute if that can help.

**Comment by helge17** [comment]
If I understand correctly, there are a few steps required to verify TuxGuitar on Flathub.

First, we need to change the app ID as described in https://github.com/flathub/flathub/issues/6395 .

What should the new app ID be? Since we are using the DNS domain tuxguitar.app, the ID must be app.tuxguitar._something_ . According to `appstreamcli validate`, ._something_ cannot be empty and should consist only of lowercase letters. In commit c47b7cb828b4fd44b7c22608ea372ec83c30ac52, I set app.tuxguitar.app as the app ID, but perhaps app.tuxguitar.tuxguitar would be better?

How is the new app ID set on Flathub? Should I just send a PR to https://github.com/flathub/ar.com.tuxguitar.TuxGuitar/ ? Will this repository then be renamed to app.tuxguitar._something_ ?

After the renaming, I will try to do the verification as described at https://docs.flathub.org/docs/for-app-authors/verification: Open an access request on GitHub, then log in at https://flathub.org/login and go through the verification process.

In parallel, there is the PR https://github.com/flathub/ar.com.tuxguitar.TuxGuitar/pull/44 for the update to 2.0.0.

**Reply by guiv42** [reply]
You're right I had forgotten this. I'll switch the PR for 2.0 update into draft state until we decide what to do

**Reply by hfiguiere** [reply]
**Renaming on flathub requires resubmitting**. And right now the package wouldn't pass a submission review.

1. build from source
2. icon must come from upstream
3. ditto for the appstream file

**Reply by guiv42** [reply]
Thank you @hfiguiere for your feedback. Nothing seems out of reach here. Could you please give us an example of a flatpak project that implements these 3 points correctly?

**Reply by helge17** [reply]
Commit 38a736e960237843c997bf88521f4ab15476c30f sets the metainfo app id to app.tuxguitar.tuxguitar, as demanded [here](https://docs.flathub.org/docs/for-app-authors/requirements#application-id):

> The ID must not end in generic terms like .desktop, .app, .linux. It's fine to repeat the application name in such cases.

For the rest, I'll have to go through the Flathub docs:

https://docs.flathub.org/docs/category/for-app-authors

But this will take some time...

**Reply by guiv42** [reply]
Shall I re-open my PR to update to 2.0.0?
We can as well re-submit another minor release.

**Reply by helge17** [reply]
> Shall I re-open my PR to update to 2.0.0?

Yes, please. It may take quite a while before we have a verified Flatpak version.
